### PR TITLE
Embed Block Bottom Sheet to set a link

### LINKED ISSUE #3274: [Embed block] Implement Bottom Sheet to set a link
### Description
- Implement Bottom Sheet to set a link for Embed block
- When a block placeholder is added, bottom sheet is automatically opened
- Link field is auto-focused (labeled “Embed link”)
- If there is a link in the clipboard it is auto-pasted
- Can type a link if empty
- Can close before typing and re-open by pressing on the embed block placeholder
- Can open by pressing the pen icon in the toolbar when an existing embed block is selected
- Can click on the “Learn more about embeds” link at the bottom to open the support article in the browser: https://wordpress.org/support/article/embeds/
- Close the bottom sheet after typing a link. If the link is not a valid URL, a warning message is shown

### Screenshot
Existing | Placeholder
-- | -- 
|

### PR DESCRIPTION
**Pull request by ceyhun**
Fixes #3274 

To test: https://github.com/WordPress/gutenberg/pull/31343

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

**Comment by cameronvoell** [comment]
Hey @ceyhun . 👋🏻 We will cut the 1.52.0 release Friday 30 April. Please let me know if you're aiming to include this PR in 1.52.0. Otherwise perhaps we can add the milestone label for 1.53.0 Thanks! 🙇

**Comment by fluiddot** [comment]
@hypest The GB-mobile PR of ["Embed Block Bottom Sheet to set a link"](https://github.com/WordPress/gutenberg/pull/31343) is ready for review, it only contains the Gutenberg ref and a release notes update, thanks 🙇 !

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 Unreleased
 ------
+* [*] Embed Block Bottom Sheet to set a link [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3436]
```
Comment (hypest) [inline RELEASE-NOTES.txt]: Oh, my understanding is that the embed block is still behind the dev flag, right? If we keep it that the release notes mention will be confusing, right?

**Comment by hypest** [comment]
> @hypest The GB-mobile PR of "Embed Block Bottom Sheet to set a link" is ready for review, it only contains the Gutenberg ref and a release notes update, thanks 🙇 !

Oops, sorry about missing to review this gb-mobile side PR. Left [a comment here](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3436#discussion_r662184210).

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (fluiddot) [reply RELEASE-NOTES.txt]: Good point, yeah it's behind a dev flag so makes sense not to include it in the release notes 👍 .

I verified if we already included entries related to the embed block and I found this one:
https://github.com/wordpress-mobile/gutenberg-mobile/blob/b5c31ba65e15680976a381016a1a722996883214/RELEASE-NOTES.txt#L39

So probably we should remove it too, wdyt?

Comment (hypest) [reply RELEASE-NOTES.txt]: Hmm, yeah, let's remove them. Technically speaking, those were already published to the Stores, but better to remove from the release notes files. Thanks for reviewing the past entries!

Comment (fluiddot) [reply RELEASE-NOTES.txt]: I've just removed all the entries in the release notes related to the embed block in [this commit](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3436/commits/867a5dc1669fc4ff71ae06bf2b4b1f17a30e790c).

**Review by hypest — APPROVED** [review]
LGTM!

**Pull request merged**